src/server/game/Battlegrounds/Battleground.cpp

```diff
@@ -1669,6 +1678,21 @@ void Battleground::PSendMessageToAll(uint32 entry, ChatMsg msgType, Player const
     va_end(ap);
 }
 
+void Battleground::AddPlayerPosition(WorldPackets::Battleground::BattlegroundPlayerPosition const position)
+{
+    _playerPositions.push_back(position);
+}
+
+void Battleground::RemovePlayerPosition(ObjectGuid guid)
+{
+    auto const& itr = std::remove_if(_playerPositions.begin(), _playerPositions.end(), [guid](WorldPackets::Battleground::BattlegroundPlayerPosition const info)
```

Comment:
Do not const& this, the type here is `std::vector<WorldPackets::Battleground::BattlegroundPlayerPosition>::iterator` - you are making a reference to temporary (gets destroyed after `;`)